vosekast_control/Tank.py

```diff
@@ -187,3 +187,8 @@ def state(self):
     def state(self, new_state):
         self._state = new_state
         self.logger.info(f"New Tank state is: {new_state}")
+
+        mqttmsg = StatusMessage(
+                    self.name, f"New Tank state is: {new_state}", None, None, None)
+        if self.mqtt.connection_test():
```

Comment:
Instat that you call every time the connection test before you send a message, you can make that test in the publish function. That is much clearer.

Reply:
moved connection_test to publish function